File: 02_ProgrammingFundamentals/_09_MapsLambdaAndStreamAPI/_01_Lecture/_02_WordSynonyms.java
package _02_ProgrammingFundamentals._09_MapsLambdaAndStreamAPI._01_Lecture;

import java.util.*;

public class _02_WordSynonyms {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // Read input value
        int countOfWords = Integer.parseInt(scan.nextLine());

        // Add variable
        Map<String, List<String>> synonymDictionary = new LinkedHashMap<>();

        // Build dictionary with synonyms
        for (int i = 0; i < countOfWords; i++) {
            String word = scan.nextLine();
            String synonym = scan.nextLine();

            if (!synonymDictionary.containsKey(word)) {
                List<String> synonymForCurrentWord = new ArrayList<>();

                synonymForCurrentWord.add(synonym);
                synonymDictionary.put(word, synonymForCurrentWord);
            } else {
                List<String> synonymsForCurrentWord = synonymDictionary.get(word);

                synonymsForCurrentWord.add(synonym);
                synonymDictionary.put(word, synonymsForCurrentWord);
            }
        }

        // Print result
        for (Map.Entry<String, List<String>> wordEntry : synonymDictionary.entrySet()) {
            String currentWord = wordEntry.getKey();
            List<String> synonymsForCurrentWord = wordEntry.getValue();

            System.out.printf("%s - %s%n", currentWord, String.join(", ", synonymsForCurrentWord));
        }

        scan.close();
    }
}

File: 24_C#Fundamentals/_07_AssociativeArrays/_01_Lecture/_02_WordSynonyms/_02_WordSynonyms.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class _02_WordSynonyms
{
    public static void Main(string[] args)
    {
        // Read input value
        int countOfWords = int.Parse(Console.ReadLine());

        // Add variable
        Dictionary<string, List<string>> synonymDictionary = new Dictionary<string, List<string>>();

        // Build dictionary with synonyms
        for (int i = 0; i < countOfWords; i++)
        {
            string word = Console.ReadLine();
            string synonym = Console.ReadLine();

            if (!synonymDictionary.ContainsKey(word))
            {
                List<string> synonymForCurrentWord = new List<string>();

                synonymForCurrentWord.Add(synonym);
                synonymDictionary[word] = synonymForCurrentWord;
            }
            else
            {
                List<string> synonymsForCurrentWord = synonymDictionary[word];

                synonymsForCurrentWord.Add(synonym);
                synonymDictionary[word] = synonymsForCurrentWord;
            }
        }

        // Print result
        foreach (KeyValuePair<string, List<string>> entry in synonymDictionary)
        {
            string currentWord = entry.Key;
            List<string> synonymsForCurrentWord = entry.Value;

            Console.WriteLine($"{currentWord} - {string.Join(", ", synonymsForCurrentWord)}");
        }
    }
}
